### ISSUE TITLE
MSI MS-5119  - PS/2 Mouse driver crashes MS-DOS

### ISSUE THREAD

**Issue by Rkccjrdss** [86Box/86Box#3684]
**Describe the bug**
Loading any DOS PS/2 mouse driver (tested  various  - CT-Mouse V1.91, 2.1beta4, Microsoft mouse V8, V9, V10, V11) causes MS-DOS 6.22, 7.1 and 8 to crash/freeze the computer and or Keyboard not to respond after driver loaded.

**To Reproduce**
Load any period correct configuration with the MSI MS-5119 motherboard (8mb, 16mb,)
2GB HD, no sound card, Intel Pentium 100mhz. PS/2 Mouse attached.
Boot to MS-Dos 6.22 and load any Dos PS/2 mouse driver.

**Expected behavior**
The mouse driver installs and the mouse works. Dos continues to function as normal.

**Desktop (please complete the following information):**
 - OS: Windows 10 Pro
 - 86Box version: [v3.11 build 4311] [ab086750e]
 - 86Box version: [v4.00 build 5000]

**Additional context**
'PS/2 Mouse support' option under 'ADVANCED CMOS SETUP' menu is set to 'ENABLED' in AMI Bios.

**Comment by borisvolk117** [comment]
I created a machine as same as yours and the same bug happens, even when I try installing windows 3.1. When win3.1 setup detects PS/2 mouse, the whole system freezes and cannot respond.

**Comment by Rkccjrdss** [comment]
> 

I hadn't tried Windows yet, but that makes sense. It appears any PS/2 mouse driver, 16bit or protected mode will crash the machine.

**Comment by OBattler** [comment]
This is now fixed - update to the latest 86Box build from Jenkins and also update your ROM set.

**Comment by Rkccjrdss** [comment]
> This is now fixed - update to the latest 86Box build from Jenkins and also update your ROM set, and delete the .BIN file from your nvr folder. It turned out to be a bug with the BIOS revision we were using, and I was lucky to be able to find an updated version that does not have this problem.

Is the 4.0 ROM Set (August 26th 2023) the latest? I has the same problem with that set.

I though it may be of interest that I found another solution to the problem by luck. 

Enter the Bios and under the menu 'PCI / PLUG AND PLAY SETUP' there is a setting called 'PCI IDE BusMaster'. When you set this to Enabled (default is disabled) any mouse driver now loads and PS/2 mouse works in Dos. 

I looked at what this setting actually does and it says it enables DMA for Hard disks  in DOS without a any driver loaded, directly via the bios.  I was skeptical about this as every motherboard I have ever used caps DOS to PIO MODE 4,  regardless of DMA settings.
I tried Norton Ghost in DOS to test this and the speed is around 900MB/minute with this setting enabled and around 300MB/minute with it disabled. So it actually works (it even allows DMA HD transfer speeds in Windows 98 without a dma driver loaded, such as in safe mode).

So very strange that disabling this option in bios (it's default setting) causes any PS/2 mouse driver to crash DOS/Windows.
